### DISCUSSION TITLE
Change github-actions config from ./ (root dir) to ./app/

### DISCUSSION THREAD

**Discussion by omqmo** [community/community#25131] [Actions]
<p>I use the configuration for <a href="https://github.com/marketplace/actions/github-pages-action#%EF%B8%8F-gatsby" rel="nofollow noopener">Gatsby</a>  from marketplace. The build is successful if the web-app project is located in root of directory (in root of github-branch). <a href="https://github.com/omqmo/gatsby-with-github-actions-example" rel="nofollow noopener">Example of working structure.</a> <a href="https://omqmo.github.io/gatsby-with-github-actions-example/" rel="nofollow noopener">Example of working minimal application on gh-pages.</a> </p>
<p>However, need to put web app in a separate app directory.<a href="https://github.com/omqmo/gatsby-with-github-actions-example/tree/next-app" rel="nofollow noopener">Example required project structure.</a> </p>
<p>Question-how do rewrite github-actions config so that build takes place with this structure?</p>

**Comment by BrightRan** [comment]
<p><a class="mention" href="/omqmo">@omqmo</a> ,</p>
<p>Do you mean you want to move files from the root directory of the repository to the " <strong>app</strong>" folder under the root directory of the repository? Have you tried using the  <strong>cp</strong> command or  <strong>mv</strong>  command to move the files?</p>
<p>Any misunderstanding, feel free to tell us.</p>
<p>If they do not work, please share your repository and more details about what you want to do with us, so that we can provide you with further help and suggestions.</p>

**Comment by omqmo** [comment]
<p><a class="mention" href="/brightran">@brightran</a> Added examples in main post.</p>

**Comment by BrightRan** [comment, marked as answer]
<p><a class="mention" href="/omqmo">@omqmo</a> ,</p>
<p>Thanks for your reply and sharing your repository.</p>
<p>On the command steps in the job, you can use  <strong>working-directory</strong>  keyword to specify the working directory of where to run the commands. And for the action steps, you just need to change the path as the relative path to the app directory.</p>
<p><img src="https://user-images.githubusercontent.com/46307996/181089077-824dec34-a519-499a-8587-abd214050a7d.png" alt="working-directory.PNG" data-base62-sha1="s9MU4I531lLNl9XU1o4MfdeFpqe" width="605" height="183"></p>
<p>For example, you can try the below configurations:</p>
<pre><code>jobs:
  deploy:
    runs-on: ubuntu-18.04
    steps:
      - uses: actions/checkout@v2

      - name: Setup Node
        uses: actions/setup-node@v1
        with:
          node-version: '12.x'

      - name: Cache dependencies
        uses: actions/cache@v1
        with:
          path: ~/.npm
          key: ${{ runner.os }}-node-${{ hashFiles('**/package-lock.json') }}
          restore-keys: |
            ${{ runner.os }}-node-

      - name: npm CI
        working-directory: ./app
        run: npm ci

      - name: npm build
        working-directory: ./app
        run: npm run build

      - name: Deploy
        uses: peaceiris/actions-gh-pages@v3
        with:
          github_token: ${{ secrets.GITHUB_TOKEN }}
          publish_dir: ./app/public</code></pre>
